### ISSUE TITLE
vsphere_virtual_machine: destroy fails: multiple virtual disks with UUID found

### ISSUE THREAD

**Issue by erbriones** [terraform-providers/terraform-provider-vsphere#404]
### Terraform Version

```
terraform version
Terraform v0.11.3
+ provider.vsphere v1.3.2
```

### vSphere Provider Version

```
terraform providers
provider.vsphere ~> 1.1
```

### Affected Resource(s)

- `vsphere_virtual_machine`

### Terraform Configuration Files

```hcl
variable "vsphere_user" {}
variable "vsphere_password" {}
variable "vsphere_server" {}
variable "vsphere_allow_unverified_ssl" {
  default = 1
}
variable "datacenter" {}
variable "cluster" {}
variable "datastore" {}
variable "image" {}
variable "network" {}
variable "network_address_list" {
  type = "list"
}
variable "network_gateway" {}
variable "network_prefix" {}
provider "vsphere" {
  user           = "${var.vsphere_user}"
  password       = "${var.vsphere_password}"
  vsphere_server = "${var.vsphere_server}"
  version        = "~> 1.1"
  # if you have a self-signed cert
  allow_unverified_ssl = "${var.vsphere_allow_unverified_ssl}"
}
data "vsphere_datacenter" "dc" {
  name = "${var.datacenter}"
}
data "vsphere_datastore" "datastore" {
  name          = "${var.datastore}"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}
data "vsphere_resource_pool" "pool" {
  name          = "${var.cluster}"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}
data "vsphere_network" "network" {
  name          = "${var.network}"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}
data "vsphere_virtual_machine" "template" {
  name          = "${var.image}"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}
resource "vsphere_folder" "example" {
  path          = "example"
  type          = "vm"
  datacenter_id = "${data.vsphere_datacenter.dc.id}"
}
resource "vsphere_virtual_machine" "vms" {
  name   = "${format("test%d", count.index + 1)}"
  folder = "${vsphere_folder.example.path}"
  datastore_id     = "${data.vsphere_datastore.datastore.id}"
  resource_pool_id = "${data.vsphere_resource_pool.pool.id}"
  num_cpus = 1
  memory   = 1024
  count    = 3
  guest_id  = "${data.vsphere_virtual_machine.template.guest_id}"
  scsi_type = "${data.vsphere_virtual_machine.template.scsi_type}"
  network_interface {
    network_id   = "${data.vsphere_network.network.id}"
    adapter_type = "${data.vsphere_virtual_machine.template.network_interface_types[0]}"
  }
  disk {
    label = "disk0"
    size  = "${data.vsphere_virtual_machine.template.disks.0.size}"
  }
  disk {
    label          = "disk1"
    size           = 10
    keep_on_remove = true
    unit_number    = 1
  }
  disk {
    label          = "disk2"
    size           = 10
    keep_on_remove = true
    unit_number    = 2
  }
  clone {
    template_uuid = "${data.vsphere_virtual_machine.template.id}"
    customize {
      linux_options {
        host_name = "${format("test%d", count.index + 1)}"
        domain    = "example.local"
      }
      network_interface {
        ipv4_address = "${element(var.network_address_list, count.index)}"
        ipv4_netmask = "${var.network_prefix}"
      }
      ipv4_gateway = "${var.network_gateway}"
    }
  }
}
```

### Expected Behavior

Expected the virtual machine resources to be destroyed.

### Actual Behavior

vsphere_virtual_machine.vms[2]: Destroying... (ID: 4204e500-3b49-11b6-10ac-e1eea3656040)
vsphere_virtual_machine.vms[0]: Destroying... (ID: 4204cd79-de2f-a945-690c-b70759803058)
vsphere_virtual_machine.vms[1]: Destroying... (ID: 42040f1e-4953-ae0c-2373-053b2a16577f)
vsphere_virtual_machine.vms.2: Still destroying... (ID: 4204e500-3b49-11b6-10ac-e1eea3656040, 10s elapsed)
vsphere_virtual_machine.vms.0: Still destroying... (ID: 4204cd79-de2f-a945-690c-b70759803058, 10s elapsed)
vsphere_virtual_machine.vms.1: Still destroying... (ID: 42040f1e-4953-ae0c-2373-053b2a16577f, 10s elapsed)
Error: Error applying plan:
3 error(s) occurred:
* vsphere_virtual_machine.vms[0] (destroy): 1 error(s) occurred:
* vsphere_virtual_machine.vms.0: disk.2: cannot find disk device: multiple virtual disks with UUID 6000C295-9c44-aba3-7c93-95d72acbc822 found
* vsphere_virtual_machine.vms[2] (destroy): 1 error(s) occurred:
* vsphere_virtual_machine.vms.2: disk.2: cannot find disk device: multiple virtual disks with UUID 6000C29d-09a7-13ba-eb6d-0a3ee6020fbf found
* vsphere_virtual_machine.vms[1] (destroy): 1 error(s) occurred:
* vsphere_virtual_machine.vms.1: disk.2: cannot find disk device: multiple virtual disks with UUID 6000C29c-309c-a02c-e1ad-52284998f080 found

### Steps to Reproduce

This is can be produced by provisioning new resources then destroying them immediately afterward. This is also an issue when using attached disks.

1. `terraform apply`
2. `terraform destroy`

**Comment by vancluever** [comment]
Hey @erbriones, thanks for the report and sorry you are having trouble. I've confirmed the issue here more than likely has something to do with `keep_on_remove` (which also makes sense as `attach` uses the same workflow for disk removal). As a workaround, it appears that removing the disk from configuration and then running `terraform apply` will still detatch the disk properly, after which the virtual machine can be destroyed normally after the last `keep_on_remove` disk is detached.

More info as we work on a fix.

Thanks!

**Comment by nwesoccer** [comment]
Ran into this same issue.  +1

**Comment by palesiak** [comment]
+1 Same issue here

**Comment by bill-rich** [comment]
Hey all. This issue should be fixed with the merging of #442. The device list was not properly being updated when detaching disks with `keep_on_remove` set to `true`. This resulted in the "multiple virtual disks with UUID" error.

**Comment by lubars** [comment]
Can confirm, now working as of vSphere plugin 1.4.0
